### PR TITLE
compute/lab: Fix typos and punctuation

### PR DESCRIPTION
**Pull request by teodorjuravlea**

### CODE REVIEW

File: content/chapters/compute/lab/content/processes.md [hunk 1]
```diff
@@ -250,11 +250,11 @@ Notice what each of the two processes prints:
 Unlike `system()`, who also waits for its child, when using `fork()` we must do the waiting ourselves.
 In order to wait for a process to end, we use the [`waitpid()`](https://linux.die.net/man/2/waitpid) syscall.
 It places the exit code of the child process in the `status` parameter.
-This argument is actually a bitfield containing more information than merely the exit code.
+This argument is actually a bit field containing more information than merely the exit code.
```
Comment (teodutu) [inline content/chapters/compute/lab/content/processes.md:253]:
```suggestion
This argument is actually a bit-field containing more information than merely the exit code.
```
I've seen it written with a dash more than separately: https://en.cppreference.com/w/cpp/language/bit_field.

File: content/chapters/compute/lab/content/processes.md [hunk 2]
```diff
@@ -250,11 +250,11 @@ Notice what each of the two processes prints:
...
 To retrieve the exit code, we use the `WEXITSTATUS` macro.
-Keep in mind that `WEXITSTATUS` only makes sens if `WIFEXITED` is true, i.e. if the child process finished on its own and wasn't killed by another one or by an illegal action (such as a seg fault or illegal instruction) for example.
+Keep in mind that `WEXITSTATUS` only makes sense if `WIFEXITED` is true, i.e. if the child process finished on its own and wasn't killed by another one or by an illegal action (such as a segfault or illegal instruction) for example.
 Otherwise, `WEXITSTATUS` will return something meaningless.
-You can view the rest of the information stored in the `status` bitfield [in the man page](https://linux.die.net/man/2/waitpid).
+You can view the rest of the information stored in the `status` bit field [in the man page](https://linux.die.net/man/2/waitpid).
```
Comment (teodutu) [inline content/chapters/compute/lab/content/processes.md:257]:
```suggestion
You can view the rest of the information stored in the `status` bit-field [in the man page](https://linux.die.net/man/2/waitpid).
```
Ditto.

File: content/chapters/compute/lab/content/threads.md [hunk 3]
```diff
@@ -64,7 +64,7 @@ The process that spawns all the others and subsequently calls `waitpid` to wait
 Update the code in `support/sum-array-bugs/seg-fault/sum_array_processes.c` and modify the call to `waitpid` to obtain and investigate this return code.
 Display an appropriate message if one of the child processes returns an error.
 
-Remember to use the appropriate [macros](https://linux.die.net/man/2/waitpid) for handling the `status` variable that is modified by `waitpid`, as it is a bitfield.
+Remember to use the appropriate [macros](https://linux.die.net/man/2/waitpid) for handling the `status` variable that is modified by `waitpid`, as it is a bit field.
```
Comment (teodutu) [inline content/chapters/compute/lab/content/threads.md:67]:
```suggestion
Remember to use the appropriate [macros](https://linux.die.net/man/2/waitpid) for handling the `status` variable that is modified by `waitpid()`, as it is a bit-field.
```

**Review by teodutu — CHANGES_REQUESTED** [review]
Thanks for the fixes. I only made a few suggestions, mainly regarding the spelling of bit-fields.

**Comment by teodorjuravlea** [comment]
Yeah, I've also looked quite a bit and couldn't find a definitive answer as to which spelling is correct. Either way, it is a bit more readable than "bitfield", so that's good.
I've added the suggestions.

**Review by teodutu — APPROVED** [review]
:+1:

**Comment by teodutu** [comment]
@teodorjuravlea I've given you 5 SMLY for this PR. Keep up the good work!

**Pull request merged**